### PR TITLE
Refactor rspec tests in moab/stanford storage_object_validator_spec

### LINKED ISSUE #145: storage_object_validator_spec tests misconstrue interaction b/w `let` statements and local vars
`storage_object_validator_spec.rb` had a bunch of local variables that were expected to interact with previously declared `let` statements, but they do not, so many of the tests are invalid.  We'll need to redraft the tests according to intent and make sure we have real coverage and not just the appearance of coverage.

### PR DESCRIPTION
**Pull request by SaravShah**
- refactoring in `spec/unit_tests/moab/storage_object_validator_spec.rb` and `spec/unit_tests/stanford/storage_object_validator_spec.rb`
- Commit by commit by easier, first commit corresponds to the `moab/sov_spec.rb` and the second commit corresponds w/ `stanford/sov_spec.rb`.

i cherry-picked (1aeaead4b39a65ea961265a3234c2c37774e79db) commit from #137 

ad-hoc paired w/ @atz  but I think your revision will still be valuable.

closes #145

### CODE REVIEW

File: spec/unit_tests/moab/storage_object_validator_spec.rb [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
   end
-  context '#validation_errors' do
-    let(:error_list) { storage_obj_validator1.validation_errors }
 
+  describe '#validation_errors' do
     it 'returns an Array of Hashes' do
-      expect(error_list).to be_kind_of Array
-      expect(error_list.first).to be_kind_of Hash
+      expect(error_list).to be_an(Array)
+      expect(error_list).to all be_a(Hash)
```
Comment (tallenaz) [inline spec/unit_tests/moab/storage_object_validator_spec.rb]: Didn't know about the `all` matcher. 👍

File: spec/unit_tests/moab/storage_object_validator_spec.rb [hunk 2]
```diff
@@ -60,61 +59,53 @@
...
-            druid_path = 'spec/fixtures/good_root01/moab_storage_trunk/gg/000/gg/0000/gg000gg0000'
-            storage_obj = Moab::StorageObject.new(druid, druid_path)
-            storage_obj_validator = described_class.new(storage_obj)
-            error_list = storage_obj_validator.validation_errors
-            expect(error_list).to be_empty
+          context 'allows multiple level sub directories' do
+            let(:druid) { 'gg000gg0000' }
+            let(:druid_path) { 'spec/fixtures/good_root01/moab_storage_trunk/gg/000/gg/0000/gg000gg0000' }
+
+            it 'does not return an error codes' do
```
Comment (tallenaz) [inline spec/unit_tests/moab/storage_object_validator_spec.rb]: "any error codes" or "an error code"?

File: spec/unit_tests/moab/storage_object_validator_spec.rb [hunk 3]
```diff
@@ -60,61 +59,53 @@
...
+                expect(error_list).to be_empty
               end
-              it 'non-forbidden subdirectories do not error' do
-                expect(allowed_subdirs_storage_obj_validator.validation_errors(true)).to eq []
+              it 'explicitly set to true does not error' do
+                expect(storage_obj_validator.validation_errors(true)).to be_empty
               end
-              it 'lack of subdirectories do not error' do
-                expect(no_subdirs_storage_obj_validator.validation_errors(true)).to eq []
+              it 'explicitly set to false and errros' do
```
Comment (atz) [inline spec/unit_tests/moab/storage_object_validator_spec.rb]: Typo here as `errros`

File: spec/unit_tests/stanford/storage_object_validator_spec.rb [hunk 4]
```diff
@@ -1,68 +1,75 @@
...
     end
-    it 'passes allow_content_subdirs argument to super' do
-      druid = 'xx000xx0000'
-      druid_path = 'spec/fixtures/bad_root01/bad_moab_storage_trunk/xx/000/xx/0000/xx000xx0000'
-      storage_object = Moab::StorageObject.new(druid, druid_path)
-      stanford_validator = described_class.new(storage_object)
-      error_list = stanford_validator.validation_errors(false)
-      expect(error_list).to include(Moab::StorageObjectValidator::CONTENT_SUB_DIRS_DETECTED => 'Version v0013: content directory should only contain files, not directories')
+    context 'allow_content_subdirs arguments' do
+      let(:druid) { 'xx000xx0000' }
```
Comment (atz) [inline spec/unit_tests/stanford/storage_object_validator_spec.rb]: These two lines can be removed, right?  Same as lines 2 and 3?

**Pull request merged**